### PR TITLE
Add a slash command example

### PR DESCRIPTION
**Pull request by thesadru**
### Summary
Added a slash command example since Hikari was lacking one.

### Checklist
- [x] I have run `nox` and all the pipelines have passed.

### CODE REVIEW

File: changes/1024.doc.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a slash command example.
```
Comment (davfsa) [inline changes/1024.doc.md:1]: No need for a changelog file :)

File: examples/slash.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# You should have received a copy of the CC0 Public Domain Dedication along with this software.
+# If not, see <https://creativecommons.org/publicdomain/zero/1.0/>.
+"""A simple bot with some simple commands."""
+
+import os
+
+import hikari
+
+bot = hikari.GatewayBot(token=os.environ["BOT_TOKEN"])
+
```
Comment (davfsa) [inline examples/slash.py:17]:
```suggestion
@bot.listen()
async def starting_event(_: hikari.StartingEvent) -> None:
    # Create commands
```

**Review by davfsa — COMMENTED** [review]
You sort of snatched this one from under my belt, had it down on my TODO list as something I need to add. But thank you regardless :)

File: examples/slash.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@bot.listen()
+async def on_interaction(event: hikari.InteractionCreateEvent) -> None:
+    """Listen for messages being created."""
```
Comment (davfsa) [inline examples/slash.py:21]:
```suggestion
async def handle_interactions(event: hikari.InteractionCreateEvent) -> None:
    """Listen for slash commands being executed."""
```

File: examples/slash.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#
+# You should have received a copy of the CC0 Public Domain Dedication along with this software.
+# If not, see <https://creativecommons.org/publicdomain/zero/1.0/>.
+"""A simple bot with some simple commands."""
+
+import os
+
+import hikari
+
+bot = hikari.GatewayBot(token=os.environ["BOT_TOKEN"])
```
Comment (davfsa) [inline examples/slash.py:16]:
```suggestion
# Set this to the guild to create the slash commands in.
# To create global commands, set to `hikari.UNDEFINED`.
#
# It is important to note that global commands can take up to an hour to show in the client
# and should only be used when "publishing" commands after they are built. For testing
# use guild based commands as they are instant. 
COMMAND_GUILD_ID = 0

bot = hikari.GatewayBot(token=os.environ["BOT_TOKEN"])
```

File: examples/slash.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    me = bot.get_me()
+    assert me is not None, "Bot hasn't been started yet"
+
+    commands = [
+        bot.rest.slash_command_builder("ping", "Get the bot's latency."),
+        bot.rest.slash_command_builder("info", "Learn something about the bot."),
+    ]
+
+    await bot.rest.set_application_commands(
+        application=me.id,
```
Comment (davfsa) [inline examples/slash.py:55]:
```suggestion
    application = await bot.rest.fetch_application()

    commands = [
        bot.rest.slash_command_builder("ping", "Get the bot's latency."),
        bot.rest.slash_command_builder("info", "Learn something about the bot."),
    ]

    await bot.rest.set_application_commands(
        application=application.id,
```

File: examples/slash.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            hikari.ResponseType.MESSAGE_CREATE,
+            f"Pong! {bot.heartbeat_latency * 1_000:.0f}ms",
+        )
+
+    elif event.interaction.command_name == "info":
+        await event.interaction.create_initial_response(
+            hikari.ResponseType.MESSAGE_CREATE,
+            "Hello, this is an example bot written in hikari!",
+        )
+
```
Comment (davfsa) [inline examples/slash.py:37]:
```suggestion
    elif event.interaction.command_name == "ephemeral":
        await event.interaction.create_initial_response(
            hikari.ResponseType.MESSAGE_CREATE,
            "Only you can see this, keep it a secret :)",
            flags=hikari.MessageFlag.EPHEMERAL,
        )
```

File: examples/slash.py (hunk 4 above)
Comment (thesadru) [reply examples/slash.py:16]: Most of the time it just works to keep it as undefined. Idk if we should completely force users to make them guild commands.

Comment (davfsa) [reply examples/slash.py:16]: I have lately seen a lot of people wondering why their commands dont show up as soon as they run. At least like this they will get a nice error that they will have to fix by deciding to either place an ID or set it to `hikari.UNDEFINED`

Comment (thesadru) [reply examples/slash.py:16]: Fair enough

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**